src/main/java/org/apache/maven/report/projectinfo/dependencies/renderer/DependenciesRenderer.java

```diff
@@ -632,6 +647,13 @@ private void renderSectionDependencyFileDetails() {
         endSection();
     }
 
+    private String getHighestVersion(double version) {
+        if (version >= 1.0d) {
```

Comment:
I think that was a bug. If all dependencies have jarDetails.getJdkRevision() == null, then the summary column would print "0.0" instead of "-". The same goes for test scope dependencies (it might be the case that there are no test scoped dependencies at all). Perhaps this is clearer: "if (version > 0.0d) ...."

Reply:
Ha, you are right! Can we move to a separate issue? I think the wiser step would be a change from `double` to `Double` and use `null` truly.

Reply:
Look at https://github.com/apache/maven-shared-jar/blob/3cd1e4d7060bba0b378c2f3bcb5bd00037b299dc/src/main/java/org/apache/maven/shared/jar/classes/JarClassesAnalysis.java#L92-L156. If classes are corrupt or JAR contains only resources that value would be `null`. Though this we can test with an minimal non-class JAR.

Reply:
I'll create a separate issue for this. However the jdk detection seems to be more complex. Currently org.codehaus.plexus:plexus-utils:3.5.0 dependency does not detect any jdk details.

Reply:
About org.codehaus.plexus:plexus-utils:3.5.0 not detecting the jdk version: It is a multi-release jar and upgrading to maven-shared-jar 3.0.0 will detect it, however it will output jdk version 10 which is wrong (compiler target is 8, with some classes with java 9 and 10 optimizations)...reporting multi-release jar dependencies still require some work.

Reply:
I have just noticed this and it is not very pleasing :-(, but that needs to be reported with Maven Shared JAR.